docs/upgrade/money-class.md

```diff
@@ -0,0 +1,430 @@
+
+If you need to set a value to the property in these custom entities or data objects directly, you should use the `Money::create()` method that accepts a string or an integer (or another [construction method](/docs/introduction/how-to-work-with-money.md#construction)).
+
+For more details read [the Money in Doctrine section](/docs/introduction/how-to-work-with-money.md#money-in-doctrine) of the documentation.
+
+## Working With Money in PHP
+You should use `Money` for all monetary values across your project.
+We strongly recommend declaring the `Money` type for both [the method arguments](http://php.net/manual/en/functions.arguments.php#functions.arguments.type-declaration) and [the return values](http://php.net/manual/en/functions.returning-values.php#functions.returning-values.type-declaration) in your methods.
+
+If you need to compute money, use its methods `add`, `subtract`, `multiply` and `divide` instead of the operators `+`, `-`, `*` and `/`.
+
+If you need to compare two instances of `Money`, use its methods `equals`, `isGreaterThan`, `isGreaterThanOrEqualTo`, `isLessThan`, `isLessThanOrEqualTo` and `compare` instead of the operators `==`/`===`, `>`, `>=`, `<`, `<=` and `<=>`.
+If you want to compare `Money` with zero, you can use the methods `isZero`, `isPositive` and `isNegative`.
+
+For more details read [the Money Class section](/docs/introduction/how-to-work-with-money.md#money-class) of the documentation.
+
+There were no changes needed in `shopsys/project-base` as most PHP implementation is in `shopsys/framework`.
+You might need to update your custom classes working with prices and overridden services, because of the changed types (and [strict typing](http://php.net/manual/en/functions.arguments.php#functions.arguments.type-declaration.strict) in some cases).
+
+*Note: Running the `phpstan` [Phing target](/docs/introduction/console-commands-for-application-management-phing-targets.md) will help you find out problems with incompatible method declarations.*
+*In case of problems, it will fail on a `PHP Fatal error` with a reference to the incompatible method (unfortunately it is unable to continue with the checks as it cannot recover from this error).*
+
+## Twig Templates
+Templates mostly use [the `price` Twig filter](/docs/introduction/how-to-work-with-money.md#price) (or one of its `price*` variants) for rendering monetary values.
+The filters work the same way, but they now require an instance of `Money` to be provided.
+If you pass a value of a different type into one of the `price*` filters (it would fail on a `TypeError`) you should check where it came from and fix it there - all monetary values have to be represented by the new `Money` value object.
+Because the monetary value is mostly taken from a `Price` object and its getters use `Money` as a return type now, the templates usually don't need any changes.
+
```

Comment:
Maybe smoke tests can help finding occurences? Not all of them, but it could help a bit

Reply:
It might help, good idea. I've added a note in e71c5b592